main.tf

```diff
@@ -626,6 +631,15 @@ resource "azurerm_kubernetes_cluster_node_pool" "node_pool" {
       condition     = can(regex("[a-z0-9]{1,8}", each.value.name))
       error_message = "A Node Pools name must consist of alphanumeric characters and have a maximum lenght of 8 characters (4 random chars added)"
     }
+    precondition {
+      condition     = (var.network_plugin_mode == "overlay" && each.value.os_type != "Windows") || (var.network_plugin_mode != "overlay")
+      error_message = "Windows Server 2019 node pools are not supported for Overlay and Windows support is still in preview"
+    }
+
+    precondition {
+      condition     = (var.network_plugin_mode == "overlay" && !regex("^Standard_DC[0-9]+s?_v2$", each.value.vm_size)) || (var.network_plugin_mode != "overlay")
```

Comment:
I made a mistake here.
https://developer.hashicorp.com/terraform/language/functions/regex

I assumed `regex` would return `false` if it did not find a match, but instead `regex` returns an error.

@lonegunmanb do you think I can fix this using `can()` ?
https://developer.hashicorp.com/terraform/language/functions/can

Reply:
Yes, `can()` function is designed for such scenarios.